src/lib.rs

```diff
@@ -94,50 +127,208 @@ fn cfg_vis_impl(
     cfg_vis_attr: CfgVisAttrArgs,
     mut item: syn::Item,
 ) -> syn::Result<proc_macro::TokenStream> {
-    fn proj_item(
-        item: &mut syn::Item,
-    ) -> syn::Result<(&mut syn::Visibility, &mut Vec<syn::Attribute>)> {
-        let proj = match item {
-            syn::Item::Const(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Enum(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::ExternCrate(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Fn(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Macro2(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Mod(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Static(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Struct(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Trait(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::TraitAlias(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Type(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Union(i) => (&mut i.vis, &mut i.attrs),
-            syn::Item::Use(i) => (&mut i.vis, &mut i.attrs),
-            _ => {
-                return Err(syn::Error::new(
-                    item.span(),
-                    "`cfg_vis` can only apply on item with visibility",
-                ));
+    let version = env!("CARGO_PKG_VERSION");
+    let __cfg_vis_accumulator_declare_path = __cfg_vis_accumulator_declare_path();
+
+    let (_, attrs) = proj_item(&mut item);
+    let mut accumulator= attrs
+        .iter_mut()
+        .filter_map(|acc_attr| {
+            if acc_attr.path == __cfg_vis_accumulator_declare_path {
+                Some((acc_attr.parse_args(), acc_attr))
+            } else {
+                None
+            }
+        })
+        .map(|(acc, attr)| {
+            let acc: CfgVisAttrArgsAccumulator = acc?;
+            if acc.version == version {
```

Comment:
Does __cfg_vis_accumulator itself need to perform such a check? E.g., only one cfg_vis and __cfg_vis_accumulator but with different versions.

Reply:
You are right